pypesto/engine/mpi_pool.py

```diff
@@ -0,0 +1,61 @@
+from multiprocessing import Pool
+from mpi4py.futures import MPIPoolExecutor
+from mpi4py import MPI
+import cloudpickle as pickle
+import logging
+from typing import List
+
+from .base import Engine
+from .task import Task
+
+
+logger = logging.getLogger(__name__)
+
+
+def work_per_node(chunk):
+    with Pool() as pool:
+        results = pool.map(work_per_core, chunk)
+    return results
+
+
+def work_per_core(pickled_task):
+    """Unpickle and execute task."""
+    task = pickle.loads(pickled_task)
+    return task.execute()
+
+
+class MPIPoolEngine(Engine):
+    """
+    Parallelize the task execution using multiprocessing.
+
+    Parameters
+    ----------
+    chunksize:
+        The chunksize the MPIPoolExecutor uses.
+        Default is two, but should always be set to
+        the number of cores used in each node.
+    """
+
+    def __init__(self, chunksize: int = 2):
+        super().__init__()
+        self.chunksize: int = chunksize
+
+    def execute(self, tasks: List[Task]):
+        """Pickle tasks and distribute work over nodes."""
+
+        pickled_tasks = [pickle.dumps(task) for task in tasks]
+        # put the tasks in batchsizes of the number of cores used
+        chunks = [pickled_tasks[x:x+self.chunksize]
```

Comment:
Indexing error if `len(tasks)` is not a multiple of chunksize?

Reply:
Checked that and if it is not a multiple, the last bit will be shorter so e.g. [1,2,3,4,5] with `chunksize=2` would be `[[1,2],[3,4],[5]]`.